### ISSUE TITLE
Missing documentation "how to run/build"

### ISSUE THREAD

**Issue by marcocastignoli** [nedpals/vex#3]
First, thanks and great work!

How do I run/build this code? Thanks :)

How do I use it as a library in my project? I tried using v install nedpals.vex but I got a series of errors :s

**Comment by nedpals** [comment]
Hi! Forgot to add additional instructions for people who are going to install vex through `vpm`. But it's just a quick change and you can start using it.

Instead of:
```v
import vex.server as vex
```
Just change it into:
```v
import nedpals.vex.server as vex
```
and it will work.

As for building/running, just like you run or compile a regular V program: `v [name of app].v` or `v run [name of app].v` .

Will add additional documentation later.

**Comment by marcocastignoli** [comment]
I did that before but I got an error there was a missing mime dependencies in the vex code, so I edited it changing to nedplas.mime, compiled but gave me a C error

**Comment by nedpals** [comment]
can you manually clone v-mime and vex without vpm? it should work properly.

**Comment by marcocastignoli** [comment]
> can you manually clone v-mime and vex without vpm? it should work properly.

Can you provide me a list of commands in order to do that?

**Comment by Delta456** [comment]
> > can you manually clone v-mime and vex without vpm? it should work properly.
> 
> Can you provide me a list of commands in order to do that?

```console
git clone https://github.com/nedpals/v-mime.git
git clone https://github.com/nedpals/vex.git
```

**Comment by marcocastignoli** [comment]
I can clone a repo :smile: I'm wondering which file I have to compile for example of this repo

**Comment by nedpals** [comment]
@marcocastignoli  If you ever stuck running the example, just do the following steps.

1. Create a folder of your app. (e.g name it as `my-vex-app`)
2. Clone `v-mime` and `vex` side-by-side inside the folder. (See Delta456's comment)
3. Create a file named `app.v` or anything that represents your vex app's code and paste the example there.
4. Execute `v run app.v` and wait for it to compile and execute.

If you ever wondered where did I get the `public/` folder and `index.html` file, just copy it from the `examples/` folder inside where the vex folder is located.

(Please see the screenshot below for the folder structure and the expected output from the terminal.)

![image](https://user-images.githubusercontent.com/7358345/69157461-e7daf680-0b1f-11ea-9afd-050b7ff01e20.png)

**Comment by marcocastignoli** [comment]
Thanks a lot for the help, I managed to make it work but I had to do these fix (I'm using V 0.1.22 1ccd197):

1) remove `or {}`
```
vex/server/server.v:59:30: `or` block cannot be applied to non-optional type
   57|  response.write('Content-Length: ${res.body.len}$separator')
   58|  response.write('Connection: close$separator')
   59|  conn.write(response.str()) or {}
```

2) replace net.MSG_PEEK with 0x02
```
warning: list.v:6 initializaing maps no longer requires `{}`
vex/server/server.v:161:78: undefined: `net.MSG_PEEK` (in module `net`)
  159|          mut line := '' // The current line segment. Can be a partial without \n in it.
  160|          for {
  161|                  n := int(C.recv(sock.sockfd, buf, HTTP_REQUEST_TYPICAL_SIZE-1, net.MSG_PEEK))
```

3) This is just a warning
You can remove {} from mime/list.v:6
```
warning: list.v:6 initializaing maps no longer requires `{}`
```

Thanks again

**Comment by nedpals** [comment]
@marcocastignoli  You should update and compile the latest V in order to fix most of the problems. Most of the changes were made to work after vlang/v#2800 was merged.

See: #2 for the related PR.
